libraries/resource_providers/api500/c7000/storage_pool_provider.rb

```diff
@@ -62,7 +61,7 @@ def update
           load_storage_system
           desired_state = Marshal.load(Marshal.dump(@item.data))
           @item.retrieve! || raise("Resource not found: The #{@resource_name} '#{@name}' could not be found")
-          update_manage_state(desired_state['isManaged'])
+          update_manage_state(desired_state['isManaged']) if desired_state['isManaged']
```

Comment:
If  ```desired_state ['isManaged']``` was false, this will never run, so I can never change to unmanaged.

Reply:
Very good, Sorry for this mistake. The correct is `unless desired_state['isManaged'].nil?`

Reply:
I think that this line is not necessary, because the action to change the management occurs with the update of StoragePool, so even with this wrong if clause, we are changing the management to true or false, always .
But maybe this line was put there with some reason, I believe that is for maintain convergence of Chef applying convergence to the status of the StoragePool management. So I will maintain that line and fix the if clause.